Fix #2409: Resolve ktlint max line in walkthrough test package

MaskedCarrot opened the pull request:
<!--
  - Thanks for submitting code to Oppia! Please fill out the following as part of
  - your pull request so we can review your code more easily.
  -->

## Explanation
Fixes #2409 
<!--
  - Explain what your PR does. If this PR fixes an existing bug, please include
  - "Fixes #bugnum:" in the explanation so that GitHub can auto-close the issue
  - when this PR is merged.
  -->

## Checklist
<!-- Please tick the relevant boxes by putting an "x" in them. -->
- [x] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] The PR explanation includes the words "Fixes #bugnum: ..." (or "Fixes part of #bugnum" if the PR only partially fixes an issue).
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary auto-generated code from Android Studio.
- [x] The PR is made from a branch that's **not** called "develop".
- [x] The PR is made from a branch that is up-to-date with "develop".
- [x] The PR's branch is based on "develop" and not on any other branch.
- [x] The PR is **assigned** to an appropriate reviewer in both the **Assignees** and the **Reviewers** sections.

MaskedCarrot committed:
linted walkthroughTest

MaskedCarrot commented:
@anandwana001 your name is not showing up in the reviewer's list.

rt4914 commented on app/src/sharedTest/java/org/oppia/android/app/walkthrough/WalkthroughActivityTest.kt:
```diff
@@ -120,7 +120,7 @@ class WalkthroughActivityTest {
 
   @Test
-  fun testWalkthroughFragment_increaseProgress_onBackPressed_decreaseProgress_progressWorksCorrectly() { // ktlint-disable max-line-length
+  fun testWalkthroughFragment_incProgress_onBackPressed_decProgress_progressWorksCorrectly() {
```
Suggested:
`fun testWalkthrough_increaseProgress_backPressed_decreaseProgress_progressIsCorrect() {`

anandwana001 replied:
```kotlin
testWalkthroughFragment_increaseProgress_pressBack_decreaseProgress_progressWorksCorrectly
```

rt4914 commented on app/src/sharedTest/java/org/oppia/android/app/walkthrough/WalkthroughFinalFragmentTest.kt:
```diff
@@ -157,7 +157,7 @@ class WalkthroughFinalFragmentTest {
 
   @Test
-  fun testWalkthroughWelcomeFragment_recyclerViewIndex2_topicSelected_configurationChanged_topicTitleIsCorrect() { // ktlint-disable max-line-length
+  fun testWalkthroughWelcomeFrag_recyclerViewIndex2_topicSelected_configChanged_topicTitleIsCrt() {
```
Suggested:
`testWalkthroughWelcome_rvIndex2_topicSelected_configChanged_topicTitleIsCorrect`

rt4914 commented on app/src/sharedTest/java/org/oppia/android/app/walkthrough/WalkthroughFinalFragmentTest.kt:
```diff
@@ -189,7 +189,7 @@ class WalkthroughFinalFragmentTest {
 
   @Test
-  fun testWalkthroughWelcomeFragment_recyclerViewIndex1_topicSelected_yesNoButton_isDisplayedCorrectly() { // ktlint-disable max-line-length
+  fun testWalkthroughWelcomeFrag_recyclerViewIndex1_topicSelected_yesNoBtn_isDisplayedCorrectly() {
```
Suggested:
`testWalkthroughWelcome_recyclerViewIndex1_topicSelected_yesNoButton_isDisplayedCorrectly`

rt4914 reviewed (changes requested):
In general replace RecyclerView with `rv` and that should sove for most cases. Also remove words like `Fragment` and `Activity`.

WDYT @anandwana001 ?

anandwana001 commented on app/src/sharedTest/java/org/oppia/android/app/walkthrough/WalkthroughFinalFragmentTest.kt:
```diff
@@ -157,7 +157,7 @@ class WalkthroughFinalFragmentTest {
 
   @Test
-  fun testWalkthroughWelcomeFragment_recyclerViewIndex2_topicSelected_configurationChanged_topicTitleIsCorrect() { // ktlint-disable max-line-length
+  fun testWalkthroughWelcomeFrag_recyclerViewIndex2_topicSelected_configChanged_topicTitleIsCrt() {
```
```kotlin
fun testWalkthroughFinalFragment_topicSelected_configurationChanged_topicTitleIsCorrect()
```

A bit correction, all the `testWalkthroughWelcomeFragment` should be `testWalkthroughFinalFragment` as this test file for that particular fragment.

MaskedCarrot replied:
done

anandwana001 commented on app/src/sharedTest/java/org/oppia/android/app/walkthrough/WalkthroughFinalFragmentTest.kt:
```diff
@@ -189,7 +189,7 @@ class WalkthroughFinalFragmentTest {
 
   @Test
-  fun testWalkthroughWelcomeFragment_recyclerViewIndex1_topicSelected_yesNoButton_isDisplayedCorrectly() { // ktlint-disable max-line-length
+  fun testWalkthroughWelcomeFrag_recyclerViewIndex1_topicSelected_yesNoBtn_isDisplayedCorrectly() {
```
we don't need `recyclerViewIndex` thing in any function name, function name just describes the action and result taken in a test.

MaskedCarrot replied:
removing this will make some function names to be  identical and since removing the word `Fragment` from the function names makes them less than 100 chars I think we should keep `recyclerViewIndex` thing

anandwana001 commented on app/src/sharedTest/java/org/oppia/android/app/walkthrough/WalkthroughFinalFragmentTest.kt:
```diff
@@ -189,7 +189,7 @@ class WalkthroughFinalFragmentTest {
 
   @Test
-  fun testWalkthroughWelcomeFragment_recyclerViewIndex1_topicSelected_yesNoButton_isDisplayedCorrectly() { // ktlint-disable max-line-length
+  fun testWalkthroughWelcomeFrag_recyclerViewIndex1_topicSelected_yesNoBtn_isDisplayedCorrectly() {
```
```kotlin
fun testWalkthroughFinalFragment_topicSelected_yesNoCardIsDisplayedCorrectly()
```

MaskedCarrot replied:
done

anandwana001 commented on app/src/sharedTest/java/org/oppia/android/app/walkthrough/WalkthroughFinalFragmentTest.kt:
```diff
@@ -212,7 +212,7 @@ class WalkthroughFinalFragmentTest {
 
   @Test
-  fun testWalkthroughWelcomeFragment_recyclerViewIndex1_topicSelected_clickNoButton_worksCorrectly() { // ktlint-disable max-line-length
+  fun testWalkthroughWelcomeFragment_recyclerViewIndex1_topicSelected_clickNoBtn_worksCorrectly() {
```
```kotlin
fun testWalkthroughFinalFragment_topicSelected_clickNoCard_progressbarIsDisplayedCorrectly()
```

MaskedCarrot replied:
done

anandwana001 commented on app/src/sharedTest/java/org/oppia/android/app/walkthrough/WalkthroughTopicListFragmentTest.kt:
```diff
@@ -151,7 +151,7 @@ class WalkthroughTopicListFragmentTest {
 
   @Test
-  fun testWalkthroughWelcomeFragment_recyclerViewIndex1_topicCard_configurationChanged_topicNameIsCorrect() { // ktlint-disable max-line-length
+  fun testWalkthroughWelcomeFrag_recyclerViewIndex1_topicCard_configChanged_topicNameIsCorrect() {
```
```kotlin
fun testWalkthroughTopicListFragment_topicCard_configurationChanged_topicNameIsCorrect()
```

please also correct other function names too as it's`testWalkthroughTopicListFragment`

MaskedCarrot replied:
should I remove the name of the fragment from the function name? since the file already specifies the name of the fragment. also, this would make all function name under 100 chars.

anandwana001 replied:
this is something mentioned in the naming convention which we have to follow- https://github.com/oppia/oppia-android/wiki/Oppia-Android-Testing#naming-convention

Although you can file a new issue for this and suggest a better naming convention and also I feel in that way we will never have this 100 char issue.

MaskedCarrot replied:
I thought of naming convention like `test_withOneCondition_withSecondCondition_hasExpectedOutcome` but this can create problems between tests like `testEmptyStringAns_emptyStringInput_exactSameStrings_verifyAnsStartsWith` and `testUppercaseStringAns_emptyStringInput_differentStrings_verifyAnsStartWith` @anandwana001 WDYT

anandwana001 replied:
File a new issue for discussion

anandwana001 reviewed (changes requested):
Thanks @MaskedCarrot 
PTAL

rt4914 commented:
@anandwana001 Your suggestions sounds good.

Luffy18346 commented:
@MaskedCarrot any updates on this?

MaskedCarrot commented:
@Luffy18346 I will file the issue by tomorrow and will continue to work on it as we discuss on the naming convention

Luffy18346 commented:
Ok Thanks

anandwana001 commented:
> @Luffy18346 I will file the issue by tomorrow and will continue to work on it as we discuss on the naming convention

Could you please resolve the conflicts and complete this PR first, before start working on naming convention.

MaskedCarrot committed:
lint

MaskedCarrot committed:
Merge branch 'develop' into resolveKtlintWalkthroughTest

7phalange7 committed:
Fix #2142: "Continue" Button increased its top/bottom padding (#2197)

* continue button increased padding fixed

* minor adjustment in previous commit

* continue button padding issue fixed for tablet

* +id changed to id in previous commit

* hardcoded the button height to 48dp

anandwana001 commented on app/src/sharedTest/java/org/oppia/android/app/walkthrough/WalkthroughActivityTest.kt:
```diff
@@ -87,15 +87,15 @@ class WalkthroughActivityTest {
 
   @Test
-  fun testWalkthroughFragment_increaseProgress_configurationChanged_worksCorrectly() {
+  fun testWalkthrough_increaseProgress_configurationChanged_worksCorrectly() {
```
use `configChange` everywhere

anandwana001 commented on app/src/sharedTest/java/org/oppia/android/app/walkthrough/WalkthroughActivityTest.kt:
```diff
@@ -71,14 +71,14 @@ import javax.inject.Singleton
 class WalkthroughActivityTest {
   @Test
-  fun testWalkthroughFragment_defaultProgress_worksCorrectly() {
+  fun testWalkthrough_defaultProgress_worksCorrectly() {
```
keep prefix unchanged everywhere for all files `testWalkthroughFragment`

anandwana001 commented on app/src/sharedTest/java/org/oppia/android/app/walkthrough/WalkthroughActivityTest.kt:
```diff
@@ -71,14 +71,14 @@ import javax.inject.Singleton
 
   @Test
-  fun testWalkthroughFragment_checkFrameLayout_backButton_progressBar_IsVisible() {
+  fun testWalkthrough_checkFrameLayout_backButton_progressBar_IsVisible() {
```
it should be camelCase `isVisible`

anandwana001 commented on app/src/sharedTest/java/org/oppia/android/app/walkthrough/WalkthroughFinalFragmentTest.kt:
```diff
@@ -107,7 +107,7 @@ class WalkthroughFinalFragmentTest {
 
   @Test
-  fun testWalkthroughWelcomeFragment_recyclerViewIndex1_topicSelected_topicTitleIsCorrect() {
+  fun testWalkthroughFinal_recyclerViewIndex1_topicSelected_topicTitleIsCorrect() {
```
as per my earlier suggestion, remove all `recyclerViewIndex` and if it brings any test name duplicate, use the text what we are testing, "firstTestTopic" as for this test.

anandwana001 commented on app/src/sharedTest/java/org/oppia/android/app/walkthrough/WalkthroughFinalFragmentTest.kt:
```diff
@@ -157,7 +157,7 @@ class WalkthroughFinalFragmentTest {
 
   @Test
-  fun testWalkthroughWelcomeFragment_rVIndex2_topicTitleIsCorrect() {
+  fun testWalkthroughFinal_recyclerViewIndex2_topicSelected_configChanged_topicTitleIsCorrect() {
```
keep consistency everywhere, use only `configChange`

anandwana001 commented on app/src/sharedTest/java/org/oppia/android/app/walkthrough/WalkthroughFinalFragmentTest.kt:
```diff
@@ -189,7 +189,7 @@ class WalkthroughFinalFragmentTest {
 
   @Test
-  fun testWalkthroughWelcomeFragment_rVIndex1_yesNoButton_isDisplayedCorrectly() {
+  fun testWalkthroughFinal_recyclerViewIndex1_topicSelected_yesNoButton_isDisplayedCorrectly() {
```
`isDisplayed` is enough

anandwana001 reviewed (changes requested):
Done a quick review 
@MaskedCarrot PTAL

anandwana001 committed:
Fix #2213: TopicListControllerTest has failing tests on CI and Fix and Format few other tests (#2216)

* fix for format tests

* nit fix

* remove exploration test from this pr

* nit fix

* ktlint fix

* nit fix

* comment added back

* wildcard import fix

anandwana001 committed:
Fix part of #2417: Fix Audio test correctness (#2425)

* fix test correctness

* nit fix

* setting progress

* added only robolectric

FareesHussain committed:
Fix Part of #2294: Migrate to TextInputLayout for reset_pin_dialog (#2349)

* issue 2098

* Migrate to TextInputLayout for reset_pin_dialog

* RTL and textInputLayout hint updated

* Ktlint fixes

* test commit

* Update gradle.xml

* Update misc.xml

* Resolving Requested changes

* Final changes

* resolving ktlint

* Fix gradle.xml

* Resolving requested changes

* Fix for disappearing error message and added test case for that

* Using Databinding

* Nit fixes in Tests

* Added suggested implementation

* NIT changes

* Completely removing ProfileInputView.kt

* Bazel build fixes

* Reverting the solution for configuration change

* Fixing the disappearing error Message in orientation change

* Resolving ktlint check

* NIT fix

* NIT fix

* Resolving Requested changes

* resolve ktlint checks

* Resolving requested changes

* NIT changes

* NIT fix

etigoel committed:
Fix #2407: Resolve ktlint max line in testing test package (#2421)

* Resolve ktlint max line in testing test package

* Resolve ktlint max line in testing test package

* Resolve ktlint max line in testing test package

* Resolve ktlint max line in testing test package

* Resolve ktlint max line in testing test package

* Resolve ktlint max line in testing test package

* Resolve ktlint max line in testing test package

* Resolve ktlint max line in testing test package

* Resolve ktlint max line in testing test package

* Resolve ktlint max line in testing test package

* Resolve ktlint max line in testing test package

* Resolve ktlint max line in testing test package

* Resolve ktlint max line in testing test package

* Resolve ktlint max line in testing test package

* Resolve ktlint max line in testing test package

* Resolve ktlint max line in testing test package

varunvjha committed:
Fix #2405 : Ktlint maxline profileprogress test package (#2444)

* resolved ktlint max line in profileprogress test package

* resolved ktlint max line profileprogress (made requested changes)

* consistency for testProfileProgressFragment (made requested changes)

* consistency for testProfileProgressFragment (made requested changes)

* made requested changes

* removed check from function names (made requested changes)

rt4914 committed:
Remove paragraph tags (#2459)

Co-authored-by: Rajat Talesra <talesra@google.com>

lakshaygarg128 committed:
Fix #2402: Resolve ktlint max line in home test package (#2424)

* RTL Support added

* Resolve ktlint max line in home test package

* resolve ktlint max line in home test package

* resolve ktlint max line in home test package and removed recycleviewindex everywhere in home

* All requested changes are done

* resolve ktlint max line in home test package

* Resolved ktlint max line in home test package

* Resolve klint max line

* Resolve klint max line

BenHenning committed:
Delete gradle.xml (#2472)

This file isn't actually needed, and is in [.gitignore](https://github.com/oppia/oppia-android/blob/da48d62/.gitignore#L7). From local testing, AS seems to regenerate gradle.xml correctly. Without removing this, people can still inadvertently make changes to gradle.xml since gitignore only applies to files not yet in the index.

FareesHussain committed:
Fix #2395: Replacing ImageView with LessonThumbnailImageView in walkthrough_topic_summary_view.xml (#2413)

* Replacing ImageView with LessonThumbnailImageView in walkthrough_topic_summary_view.xml

* Added lesson tag

* Injected EntityType

* Added test for image

* NIT fix

* ktlint fix

* NIT fix

BenHenning committed:
CODEOWNERS CI validation + pattern fixes (#2474)

* Another try at fixing layout file ownership + CI

It seems the previous layout pattern wasn't actually matching to @rt4914. I'm hoping the adjustment fixes the issue.

Also, I'm going to try removing my global ownership & instead rely on a CI check to verify that new files that aren't owned are properly checked (https://github.com/mszostok/codeowners-validator).

* Create static_checks.yml

Create new workflow for static analysis checks (eventually we can move our linter checks to these to keep our tests in separate, dedicated workflows).

* Update static_checks.yml

Check out the repository before testing CODEOWNERS.

* Update main.yml

Temporarily disable other CI actions to avoid noise while the static checks workflow is debugged.

* Update CODEOWNERS

Remove extra & incorrect 'res' from resource file path.

* Update CODEOWNERS

Add files missing corresponding code owners, and try to blanket cover resources from all modules.

* Update CODEOWNERS

Use more specific resource file patterns since Git/GitHub doesn't like the more generic version.

* Update CODEOWNERS

Remove redundant lines (since presumably they aren't actually having the desired effect).

* Update CODEOWNERS

Fix incorrect patterns (thanks to CI check!).

* Update CODEOWNERS

Try adding invalid syntax to trigger a specific CI validation failure.

* Update CODEOWNERS

Remove intentional syntax error.

* Create random_file

Add random file to verify that CI checks catch this file doesn't have corresponding owners.

* Delete random_file

Test completed.

* Update main.yml

Re-enable main workflows now that the static checks workflow is finalized.

* Update CODEOWNERS

Move PNG file codeowners to me since they count as binary files and we want to avoid checking those into the root repository as much as possible.

MaskedCarrot commented:
@anandwana001 since the issue of function name under 100 chars is already resolved in #2408 so this PR no longer solves that issue. But the function names are not correct in `walkthroughFinalFragmentTest` as you pointed out earlier. Should i close issue #2409 and file a new issue to correct function names and continue working there?

anandwana001 commented:
It looks like the prefix issue is not yet fixed, so I suggest to update this pr with the latest pull from develop branch and continue work in this PR only.

jcqli committed:
Fix #2431: Upgrade kotlin to 1.4 (#2439)

* Upgrade kotlin in gradle build, update moshi dependency in data module

* Change bazel kotlin rules to 1.5.0

* Use 1.5.0 alpha 2 and add SHA

* Add rules_java http_archive

* Add rules_proto to migrate from native to Starlark proto rules

* Go back to kotlin 1.3 to check rules_proto

* Revert "Go back to kotlin 1.3 to check rules_proto"

This reverts commit 6f2c498162300f1f58467f243b27012ad0747cda.

* Add kt_download_local_dev_dependencies() in WORKSPACE

* Address review comments

* Remove rules_java import

* Fix comment

* Address review comments

* Import rules_java for java_lite_proto_library

* Go back to gradle 3.6

* Alphabetize

* Use Kotlin 1.3.61?

* Revert "Use Kotlin 1.3.61?"

This reverts commit 611eb41a14129d5c143c9ef8b158cdb93b5d8a9c.

* Use gradle 6.8, Kotlin 1.4, plugin 3.6

* Use Kotlin 1.3 and compatibility with Gradle and plugin

* Kotlin 1.4, gradle 3.6, wrapper 6.8

MaskedCarrot committed:
fixed fun names

MaskedCarrot committed:
Merge branch 'resolveKtlintWalkthroughTest' of https://github.com/MaskedCarrot/oppia-android into resolveKtlintWalkthroughTest

MaskedCarrot committed:
fixed function names

anandwana001 commented:
@MaskedCarrot It looks like the merge is not correctly happened. Could you please fix it or if it takes time, you can close this R and create a new one.
